### PR TITLE
Added electronic journal to the transaction search

### PR DESCRIPTION
**Pull request by blubaugha**
### Issues Fixed
[#2243](https://github.com/JumpMind/commerce/issues/2243)

### Summary
This adds an **Electronic Journal**, to the transaction search screen, that allows users to search all transaction types. Also, the **View Details** button on `POST_VOID` transaction types will navigate to the original voided transaction.

### Parameter Changes
There is a configuration option to enable/disable this feature. It can be used to enable this for the BO and disable on the register.
```
openpos:
  transactionSearch:
    DOC__electronicJournal: "Enables the electronic journal."
    enableElectronicJournal: true
```

### Screenshots
**Post Void -> Navigates to original voided transaction**

trans-search--desktop-post-void

**Disable Electronic Journal**

trans-search--disable-electronic-journal

**Desktop**

trans-search--desktop

**ELO POS - Landscape**

trans-search--elopos-landscape

**ELO POS - Portrait**

trans-search--elopos-portrait

**iPad Pro - Landscape**

trans-search--ipadpro-landscape

**iPad Pro - Portrait**

trans-search--ipadpro-portrait

**iPad - Landscape**

trans-search--ipad-landscape

**iPad - Portrait**

trans-search--ipad-portrait

### CODE REVIEW

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java [hunk 1]
```diff
@@ -68,5 +68,17 @@ public static long daysBetween(Date date1, Date date2) {
...
+    public static String changeFormat(String value, String existingFormat, String newFormat) {
+        if(StringUtils.isBlank(value)) {
+            return value;
+        }
+
+        try {
+            Date date = new SimpleDateFormat(existingFormat).parse(value);
+            return new SimpleDateFormat(newFormat).format(date);
+        } catch (ParseException e) {
+            return value;
```
Comment (jrmihalick) [inline openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:81]: Should log a warn message here

**Review by jrmihalick — CHANGES_REQUESTED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java (hunk 1 above)
Comment (blubaugha) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:85]: Fixed

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java [hunk 2]
```diff
@@ -78,6 +81,7 @@ public static String changeFormat(String value, String existingFormat, String ne
             Date date = new SimpleDateFormat(existingFormat).parse(value);
             return new SimpleDateFormat(newFormat).format(date);
         } catch (ParseException e) {
+            logger.warn("Unable to change format of date \"{}\" from \"{}\" to \"{}\"", value, existingFormat, newFormat, e);
```
Comment (jrmihalick) [inline openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java:84]: That won't log the exception stack trace.  To do that, needs to be something like this:

```logger.warn(String.format("Unable to change format of date '%s' from '%s' to '%s'", value, existingFormat, newFormat), e);```

**Review by jrmihalick — CHANGES_REQUESTED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java (hunk 2 above)
Comment (blubaugha) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java]: Haha no worries - yeah, I intended to log the stack trace, so thanks for the catch

**Review by jrmihalick — APPROVED** [review]

File: openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java (hunk 2 above)
Comment (mmichalek) [reply openpos-service/src/main/java/org/jumpmind/pos/service/util/DateUtils.java]: Nice pun there Andy.  "the catch'

**Review by mmichalek — APPROVED** [review]

**Pull request merged**